src/main/java/org/primefaces/util/LangUtils.java

```diff
@@ -415,6 +415,30 @@ public static String md5Hex(byte[] bytes) {
         }
     }
 
+    /**
+     * Converts a sting to capital case even counting Unicode characters.
+     * <pre>
+     * thisIsMyString = This Is My String
+     * ThisIsATest = This Is A Test
+     * helloWorld = Hello World
+     * </pre>
+     *
+     * @param value the value to capital case
+     * @return the returned value in capital case or empty string if blank
+     */
+    public static String toCapitalCase(String value) {
+        if (LangUtils.isValueBlank(value)) {
+            return Constants.EMPTY_STRING;
+        }
+
+        // split on unicode uppercase characters
+        String[] values = value.split("(?<=.)(?=\\p{Lu})");
```

Comment:
Use pre-compiled regex

Reply:
I can't believe I missed that!   Thanks its fixed.